src/typ/expr.rs

```diff
@@ -1523,6 +1524,8 @@ impl<'a, 'b, 'c> ExprTyper<'a, 'b, 'c> {
                 let tag = match &constructor.variant {
                     ValueConstructorVariant::Record { name, .. } => name.clone(),
 
+                    ValueConstructorVariant::Inline => todo!(),
```

Comment:
Let's return a new error here, one saying that inline types cannot be used in clause guards. We can do the same for the other TODO in this function.

Later we add support for this if it needed, but it would be a bigger job.